tests/all/imports.rs

```diff
@@ -668,3 +671,40 @@ fn custom_type() {
         "#)
         .test();
 }
+
+#[test]
+fn unused_imports_not_generated() {
+    project()
+        .debug(false)
+        .file("src/lib.rs", r#"
+            #![feature(proc_macro, wasm_custom_section, wasm_import_module)]
+
+            extern crate wasm_bindgen;
+
+            use wasm_bindgen::prelude::*;
+
+            #[wasm_bindgen]
+            extern {
+                pub fn foo();
+            }
+
+            #[wasm_bindgen]
+            pub fn run() {
+            }
+        "#)
+        .file("test.ts", r#"
+            import { run } from "./out";
+
+            export function test() {
+                run();
+            }
+        "#)
+        .test();
+
+    let out = ::root().join("out.js");
+    let mut contents = String::new();
+    File::open(&out).unwrap()
+        .read_to_string(&mut contents).unwrap();
```

Comment:
We have a helper function for reading `out.js` now :)

```rust
let project = project()
    ...

let contents = project.read_js();
```

Reply:
Oh nice!